commands/create_kubeconfig.go

```diff
@@ -44,153 +89,205 @@ func init() {
+
+		switch {
+		case IsCouldNotCreateClientError(err):
+			headline = "Failed to create API client."
+			subtext = "Details: " + err.Error()
+		case util.IsCouldNotSetKubectlClusterError(err):
+			headline = "Error: " + err.Error()
+			subtext = "API endpoint would be: " + result.apiEndpoint
+			subtext += "\nCA file path would be: " + result.caCertPath
+		case util.IsCouldNotSetKubectlCredentialsError(err):
+			headline = "Error: " + err.Error()
+			subtext = "Client key path would be: " + result.clientKeyPath
+			subtext += "\nClient certificate path would be: " + result.clientCertPath
+		case util.IsCouldNotSetKubectlContextError(err):
+			headline = "Error: " + err.Error()
+		case util.IsCouldNotUseKubectlContextError(err):
+			headline = "Error: " + err.Error()
+			subtext = "Context name to select is: giantswarm-" + args.clusterID
+		case IsClusterNotFoundError(err):
+			headline = fmt.Sprintf("Error: Cluster '%s' does not exist.", args.clusterID)
+			subtext = "Please check the ID spelling or list clusters using 'gsctl list clusters'."
+		default:
+			headline = err.Error()
+		}
+
+		// Print error output
+		fmt.Println(color.RedString(headline))
+		if subtext != "" {
+			fmt.Println(subtext)
+		}
+		os.Exit(1)
+	}
+
+	// Success output
+
+	fmt.Println("Creating new key pair…")
+
+	fmt.Printf("New key pair created with ID %s and expiry of %v hours\n",
+		util.Truncate(util.CleanKeypairID(result.keypairResponse.Id), 10),
+		result.keypairResponse.TtlHours)
+
+	fmt.Println("Certificate and key files written to:")
+	fmt.Println(result.caCertPath)
+	fmt.Println(result.clientCertPath)
+	fmt.Println(result.clientKeyPath)
+
+	fmt.Printf("Switched to kubectl context 'giantswarm-%s'\n\n", args.clusterID)
+
+	// final success message
+	fmt.Println(color.GreenString("kubectl is set up. Check it using this command:\n"))
+	fmt.Println(color.YellowString("    kubectl cluster-info\n"))
+	fmt.Println(color.GreenString("Whenever you want to switch to using this context:\n"))
+	fmt.Println(color.YellowString("    kubectl config use-context giantswarm-%s\n", args.clusterID))
+
+}
+
+func createKubeconfig(args createKubeconfigArguments) (createKubeconfigResult, error) {
+	result := createKubeconfigResult{}
 
 	clientConfig := client.Configuration{
-		Endpoint:  endpoint,
+		Endpoint:  args.apiEndpoint,
 		Timeout:   10 * time.Second,
 		UserAgent: config.UserAgent(),
 	}
 	apiClient, clientErr := client.NewClient(clientConfig)
 	if clientErr != nil {
-		fmt.Println(color.RedString("Error: %s", clientErr.Error()))
-		os.Exit(1)
+		return result, microerror.Maskf(couldNotCreateClientError, clientErr.Error())
 	}
-	authHeader := "giantswarm " + token
+
+	authHeader := "giantswarm " + args.authToken
 
 	// get cluster details
-	clusterDetailsResponse, apiResponse, err := apiClient.GetCluster(authHeader, cmdClusterID, requestIDHeader, createKubeconfigActivityName, cmdLine)
+	clusterDetailsResponse, apiResponse, err := apiClient.GetCluster(authHeader,
```

Comment:
I would break this more like
```
clusterDetailsResponse, apiResponse, err := apiClient.GetCluster(
   authHeader,
   args.clusterID,
   requestIDHeader,
.. etc
```

Reply:
done